### PR TITLE
[RW-8056][risk=no] Added new API for Universal Search call

### PR DESCRIPTION
**Pull request by chenchals**
Description:
- new API call:`/criteria/ehrDomainCounts` controller call: findEhrDomainCounts
- new service call: findEhrDomainCounts
- calls existing service/dao methods
- Call cant be tested since the underlying SQL query is a `nativeQuery`

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java [hunk 1]
```diff
@@ -369,18 +369,22 @@ public Long findDomainCountByStandard(String domain, String term, Boolean standa
...
+  public List<CardCount> findEhrDomainCounts(String term) {
+    return findDomainCounts(
+        term,
+        true,
+        ImmutableList.of(
+            Domain.CONDITION,
+            Domain.DRUG,
+            Domain.MEASUREMENT,
+            Domain.OBSERVATION,
+            Domain.PROCEDURE));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java:382]: Sorry, missed this before. List needs to include Visit and Devices

**Review by freemabd — COMMENTED** [review]
Looks good, just one comment below.

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java (hunk 1 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java:382]: ok got it.

Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java:382]: no particular order, right?

Comment (freemabd) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java:382]: The order of the cards for the UI is in the domain_card table sort_order column.

Comment (freemabd) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java:382]: but for Universal search we may want to return them in order by highest count

Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java]: done!

**Review by freemabd — APPROVED** [review]

**Pull request merged**